draft-ietf-anima-brski-prm.md

```diff
@@ -520,127 +540,148 @@ In {{!RFC8995}}, pledges instead need to continuously request enrollment from a
+For this, it MUST possess the IDevID trust anchor(s) (i.e., root or issuing CA certificate(s)) of the pledge vendor(s)/manufacturer(s).
 This behavior is retained in BRSKI-PRM.
 
-The domain registrar MUST possess and trust the IDevID (root or issuing) CA certificate 
-of the pledge vendor/manufacturer.
+In its role as EST server {{!RFC7030}}, the domain registrar MUST also possess the domain CA certificates as defined in {{Section 5.9 of !RFC8995}}.
+
+Finally, the domain registrar MUST possess the Registrar-Agent EE certificate(s) to validate agent-signed data and to provide it to the MASA.
+The registrar MAY use the certificate verified during client authentication within the TLS sessions with the Registrar-Agent;
+in this case, the registrar MUST possess the domain trust anchor (i.e., domain CA certificate) for the Registrar-Agent EE certificate to verify the certificate chain.
+Alternatively, the Registrar-Agent EE certificate(s) MAY be provided via configuration or a repository.
 
-Further, the domain registrar MUST have its own EE credentials.
 
 ### Domain Registrar with Combined Functionality
 
-A registrar with combined BRSKI and BRSKI-PRM functionality MAY detect if the bootstrapping is performed by the pledge directly (BRSKI case) or by a Registrar-Agent (BRSKI-PRM case) based on the utilized credential for client authentication during the TLS session establishment and switch switch the operational mode from BRSKI to BRSKI-PRM.
+A registrar with combined BRSKI and BRSKI-PRM functionality MAY detect if the bootstrapping is performed by the pledge directly (BRSKI case) or by a Registrar-Agent (BRSKI-PRM case) based on the utilized credentials for client authentication during the TLS session establishment and switch the operational mode from BRSKI to BRSKI-PRM.
+
+This may be supported by a specific naming in the SAN (subject alternative name) component of the Registrar-Agent EE certificate. TODO(totally unclear purpose)
+
+Alternatively, this may be supported by using an EE certificate signed by the domain owner for the client authentication of the Registrar-Agent. TODO(statement in l.602 already relies on this EE cert, what is the alternative here?)
+Using an EE certificate also allows the registrar to verify that a Registrar-Agent is authorized to perform the bootstrapping of a pledge. TODO(yes, this is why there is an RegAgt EE cert in the first place; what is the point here?)
+See also Agent Proximity Assertion in {{agt_prx}}.
 
-This may be supported by a specific naming in the SAN (subject alternative name) component of the EE certificate of the Registrar-Agent.
+Using an EE certificate for TLS client authentication of the Registrar-Agent is a deviation from {{!RFC8995}}, in which the pledge IDevID certificate is used to perform TLS client authentication. TODO(again l.602 is already based on this fact.)
```

Comment:
This and the other two TODOs above all come from the question what additional information is given in this text.
Line 602 already clearly states how to distinguish, which is again said in Section 7.3:

> Consequently, the domain registrar can distinguish BRSKI (pledge-initiator-mode) from BRSKI-PRM (pledge-responder-mode).

Reply:
This is rather a note that this step is a deviation from BRSKI. Does it become clearer with the changes from the TODOs above?

Reply:
Reworked with important statments only.